src/Multipart/AbstractUploader.php

```diff
@@ -19,6 +19,8 @@
     public function __construct(Client $client, $source, array $config = [])
     {
         $this->source = $this->determineSource($source);
+        $config['exception_params']['file_name'] = $this->source->getMetadata('uri');
```

Comment:
That's correct, and makes perfect sense to me, thanks Michael. I fixed it and did renaming in the latest commit.

I have also done some manually testing with "real" Exceptions and checked this work as expected :)